--- ../out/libraries/okhttp-gson/modelInnerEnum.mustache	2025-08-11 14:22:51.473032410 +0200
+++ modelInnerEnum.mustache	2025-08-11 14:24:09.583029710 +0200
@@ -43,7 +43,7 @@
           return b;
         }
       }
-      {{#isNullable}}return null;{{/isNullable}}{{^isNullable}}{{#enumUnknownDefaultCase}}{{#allowableValues}}{{#enumVars}}{{#-last}}return {{{name}}};{{/-last}}{{/enumVars}}{{/allowableValues}}{{/enumUnknownDefaultCase}}{{^enumUnknownDefaultCase}}throw new IllegalArgumentException("Unexpected value '" + value + "'");{{/enumUnknownDefaultCase}}{{/isNullable}}
+      return null;
     }
 
     public static class Adapter extends TypeAdapter<{{{datatypeWithEnum}}}{{^datatypeWithEnum}}{{classname}}{{/datatypeWithEnum}}> {
